scripts/apps/authoring/authoring/services/AuthoringService.ts

```diff
@@ -264,6 +266,32 @@ export function AuthoringService($q, $location, api, lock, autosave, confirm, pr
             );
     };
 
+    this.unpublish = function unpublish(item) {
```

Comment:
Can you add a type `unpublish(item: IPublishedArticle)` and validation to ensure that `item` is really published?

Reply:
Which validation? Before everything?

Reply:
Using this function before? `isIPublishedArticle()`

Reply:
Yep, you can use this function to. It would also be good to log the error like

```
if(isIPublishedArticle(item) !== true) {
    logger.warn(...);
    return;
}

... unpublishing logic
```